### PR TITLE
Bugfix/coin tester setup

### PR DESCRIPTION
**Pull request by lvndry**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 

### 📝 Description

Repare polkadot coin tester

image

### ❓ Context

- **JIRA or GitHub link**: N/A

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/coin-modules/coin-polkadot/src/network/node/apiPromise.ts [hunk 1]
```diff
@@ -11,10 +12,22 @@ export default async function () {
...
       ? { Authorization: "Basic " + coinConfig.node.credentials }
       : undefined;
-    api = await ApiPromise.create({
-      provider: new HttpProvider(coinConfig.node.url, headers),
-      noInitWarn: true, //to avoid undesired warning (ex: "API/INIT: polkadot/1002000: Not decorating unknown runtime apis")
-    });
+
+    const nodeURL = coinConfig.node.url;
+
+    if (nodeURL.startsWith("ws://") || nodeURL.startsWith("wss://")) {
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-polkadot/src/network/node/apiPromise.ts:18]: Just extract the provider in a separate variable and keep out of the "if scope" the creation of the ApiPromise.

File: libs/coin-modules/coin-polkadot/src/test/coin-tester/docker-compose.yml [hunk 2]
```diff
@@ -17,8 +18,8 @@ services:
         retries: 12
   sidecar-api:
     image: ghcr.io/ledgerhq/polkadot-sidecar:19.0.0-ledger-live
-    platform: linux/x86_64
     container_name: sidecar-api
+    platform: linux/x86_64
```
Comment (sprohaszka-ledger) [inline libs/coin-modules/coin-polkadot/src/test/coin-tester/docker-compose.yml:22]: It will work on a MacOS?

**Review by sprohaszka-ledger — COMMENTED** [review]
2 small feedback

File: libs/coin-modules/coin-polkadot/src/test/coin-tester/docker-compose.yml (hunk 2 above)
Comment (lvndry) [reply libs/coin-modules/coin-polkadot/src/test/coin-tester/docker-compose.yml:22]: It's actually a fix for macos

**Review by sprohaszka-ledger — APPROVED** [review]
👍

**Review by hzheng-ledger — APPROVED** [review]
looks good to me

**Pull request merged**